File: public/projects/update_html.js
const fs = require('fs');

try {
    let html = fs.readFileSync('index.html', 'utf8');

    // 1. Extract timeline-items
    const timelineMatch = html.match(/<div class="timeline-items">([\s\S]*?)<\/div>\s*<\/section>/);
    if (!timelineMatch) {
        console.error("Could not find timeline-items");
        process.exit(1);
    }
    const timelineContent = timelineMatch[1];

    // 2. Remove old video-portfolio section
    // Starts with: <section class="portfolio video-portfolio"
    // Ends with: </section> where timeline items was
    html = html.replace(/\s*<section class=\"portfolio video-portfolio\" id=\"video-portfolio\" style=\"display: none;\">[\s\S]*?<\/div>\s*<\/section>/, '');

    // 3. Replace the service box
    const oldBoxRegex = /<div class="service-box" onclick="showVideoPortfolio\(\)">\s*<div class="service-info">\s*<h4>Video Editing<\/h4>\s*<p>Turn raw footage into compelling stories with professional editing that\s*enhances every frame\.\s*<\/p>\s*<\/div>\s*<\/div>/;

    const newBox = `<div class="service-box video-editing-box" onclick="toggleAccordion(this, event)">
                    <div class="service-info">
                        <h4>Video Editing</h4>
                        <p>Turn raw footage into compelling stories with professional editing that
                            enhances every frame.
                        </p>
                        <span class="expand-icon"><i class='bx bx-chevron-down'></i></span>
                    </div>
                    <div class="accordion-content-wrapper">
                        <div class="accordion-content">
                            <div class="scroll-timeline-container">
                                <div class="scroll-dot"></div>
                                <div class="scroll-line-bg"></div>
                                <div class="scroll-line-fill" id="scroll-line-fill"></div>
                            </div>
                            <div class="timeline-items">
${timelineContent}
                            </div>
                        </div>
                    </div>
                </div>`;

    if (oldBoxRegex.test(html)) {
        html = html.replace(oldBoxRegex, newBox);
        fs.writeFileSync('index.html', html, 'utf8');
        console.log('Successfully updated index.html');
    } else {
        console.error('Could not find exact oldBox string to replace');
    }

} catch (err) {
    console.error(err);
}

File: public/projects/update_html.py
import re

def main():
    try:
        with open('index.html', 'r', encoding='utf-8') as f:
            html = f.read()

        # 1. Extract the timeline-items inner HTML
        match = re.search(r'<div class=\"timeline-items\">(.*?)</div>\s*</section>', html, re.DOTALL)
        if not match:
            print('Could not find timeline-items')
            return
        timeline_content = match.group(1)

        # 2. Remove the old video-portfolio section completely
        html = re.sub(r'\s*<section class=\"portfolio video-portfolio\" id=\"video-portfolio\" style=\"display: none;\">.*?</section>', '', html, flags=re.DOTALL)

        # 3. Replace the Video Editing service-box
        old_box = '''<div class="service-box" onclick="showVideoPortfolio()">
                    <div class="service-info">
                        <h4>Video Editing</h4>
                        <p>Turn raw footage into compelling stories with professional editing that
                            enhances every frame.
                        </p>
                    </div>
                </div>'''

        new_box = f'''<div class="service-box video-editing-box" onclick="toggleAccordion(this, event)">
                    <div class="service-info">
                        <h4>Video Editing</h4>
                        <p>Turn raw footage into compelling stories with professional editing that
                            enhances every frame.
                        </p>
                    </div>
                    <div class="accordion-content-wrapper">
                        <div class="accordion-content">
                            <div class="scroll-timeline-container">
                                <div class="scroll-dot"></div>
                                <div class="scroll-line-bg"></div>
                                <div class="scroll-line-fill" id="scroll-line-fill"></div>
                            </div>
                            <div class="timeline-items">
                                {timeline_content}
                            </div>
                        </div>
                    </div>
                </div>'''

        if old_box in html:
            html = html.replace(old_box, new_box)
            with open('index.html', 'w', encoding='utf-8') as f:
                f.write(html)
            print('Successfully updated index.html')
        else:
            print('Could not find the target old_box exactly, skipping replace')

    except Exception as e:
        print(f"Error: {e}")

if __name__ == '__main__':
    main()
